reports/spec4.tex

```diff
@@ -2,7 +2,36 @@
 
 \section{M3: Spec 4}
 
-TODO\footnote{JK: inductive invariant}
+\subsection{Inductive Invariant} 
+This specification defines an inductive invariant \textsf{IndInv}. Recall, an inductive invariant satisfies the following two conditions, for a spec with an initial-state predicate \textsf{Init}, and transition predicate \textsf{Next}:
+\begin{enumerate}
+	\item It is implied by the initial state: $\mathsf{Init} \Rightarrow \mathsf{IndInv}$
+	\item It is preserved by the transition predicate: $\mathsf{IndInv} \land \mathsf{Next} \Rightarrow \mathsf{IndInv}$
+\end{enumerate}
+Because of this characterization, inductive invariants lend themselves especially nicely to bounded symbolic model checking; with Apalache, one can prove (or disprove) an inductive invariant by running two queries of depth at most 1, corresponding to the above properties.
+If no violation is found, we are assured that \textsf{IndInv} holds in all possible reachable states.
+Since our goal is ultimately to prove or disprove \textsf{AccountableSafety}, we can additionally prove $\textsf{IndInv} \Rightarrow \mathsf{AccountableSafety}$.
+
+The challenge, typically, is that inductive invariants are more difficult to write, compared to state invariants.
+They are usually composed of several lemmas, i.e. properties that we are less interested in on their own, but which are crucial in establishing property (2.).
+
+The invariant introduced in Spec~4 mainly deals with the characterization of the chain-fork scenarios. 
```

Comment:
> The invariant introduced in Spec~4 mainly deals with the characterization of the chain-fork scenarios. 

This is a bit confusing, because you say here it mainly deals with chain forks, but then there is a whole other section about justified checkpoints.

Maybe preface it in another way

```suggestion
As we explain below, the inductive invariant introduced in \SpecFour{} mainly contains of two sets of lemmas, one for characterizing the chain-fork scenarios, and a second one for characterizing justified checkpoints.
```